src/Microsoft.Sbom.Api/Executors/ComponentDetectionBaseWalker.cs

```diff
@@ -107,8 +113,147 @@ async Task Scan(string path)
 
             var uniqueComponents = FilterScannedComponents(scanResult);
 
+            var listOfLicenses = new List<string>();
+            var clearlyDefinedType = string.Empty;
+            var clearlyDefinedProvider = string.Empty;
+            var clearlyDefinedNamespace = string.Empty;
+            var clearlyDefinedName = string.Empty;
+            var clearlyDefinedRevision = string.Empty;
+            // var clearlyDefinedExpand = "-files";
+
+            List<string> listOfComponentsForApi = new List<string>();
```

Comment:
I think most of this new code makes sense in its own class (maybe `ILicenseFinder` with implementation `ClearlyDefinedLicenseFinder` or some such [I don't love "Finder" but am failing for other ideas]) which you'll create as a singleton, then you can give it the metadata here and have it do all the logic about weather it needs to hit the API or not internally. Creating it as a singleton also might save you from having a public static Dictionary.

Reply:
Thanks, this sounds a lot cleaner than the current approach.